api/v1alpha1/virtualmachine_types.go

```diff
@@ -106,6 +106,10 @@ const (
 	// marked userConfigurable and already present in either OVF Properties of a VirtualMachineImage
 	// or as vApp properties on an existing VM or VMTX will be set, all others will be ignored.
 	VirtualMachineMetadataOvfEnvTransport VirtualMachineMetadataTransport = "OvfEnv"
+
+	// VirtualMachineMetadataCloudInitPrepTransport will set the VirtualMachineMetadata ConfigMap data as
```

Comment:
Can we please rename this to:

```golang
VirtualMachineMetadataCloudInitTransport VirtualMachineMetadataTransport = "CloudInit"
```

We do not really need to reveal more than this is cloud-init IMO. In the future I would even like to rename the type to `VirtualMachineBootstrapRef` and let it point directly to an object reference, with that determining the provider. We need to move away from `ConfigMap` to `Secret` at some point to shield possibly sensitive data. For now let's just remove the specifics of the actual transport beyond saying `CloudInit`. Thanks!

Reply:
Generally I'm in favor of making things generic as well, but in this case, being specific/explicit about the underlying transport is preferable imo. Also, we need some way to differentiate between CloudInitPrep vs. upcoming CloudInitNoCloud.

Reply:
> Generally I'm in favor of making things generic as well, but in this case, being specific/explicit about the underlying transport is preferable imo. Also, we need some way to differentiate between CloudInitPrep vs. upcoming CloudInitNoCloud.

Is this important to the consumer of the API or the implementation? Is there any penalty for changing how the cloud-init data is transported?

Reply:
Even at the implementation level, this would help us in differentiating between the two.

Reply:
> Even at the implementation level, this would help us in differentiating between the two.

When would we use two? Why would we need to do so? Also, more importantly, why would we expect someone interacting with the API to know which to choose?

Reply:
Once we support NoCloud by dynamically building an ISO and mounting it to the VM, the need for any other datasource, such as Cloud Init Prep, is greatly diminished. The one exception _might_ be on VMC. Originally you could not use the NoCloud DS on VMC because you could not upload an ISO to a datastore on VMC and mount it to a VM. If this is still the case, we would likely need to use dual cloud-init transports.

However, I think if that need arises, that is a decision the VM Operator can and should make without requiring input from the user. For example, a flag that indicates VM Operator is deployed on VMC.

Reply:
If specifying underlying transport to the customers is important then we need cloudInitPrep. Otherwise, how about having cloudInit as a generic name and if later we end up having both kinds, implementation details will remember that cloudInit means cloudInitPrep. But if we're switching to cloudInitNoCloud completely, then this won't be a problem.